### ISSUE TITLE
Loss functions

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire-ml#34]
- [x] mean squared error
- [x] mean absolute error
- [x] binary cross entropy
- [ ] negative log likelyhood

**Comment by syurkevi** [comment]
https://github.com/arrayfire/arrayfire-ml/pull/36#issuecomment-317360576
ArrayFire has lookup, however its behavior seems a bit weird, would it need to be changed in ArrayFire or just emulated in autograd?

**Comment by 9prady9** [comment]
@syurkevi What kind of behavior difference are you talking about ? Can you show me an example ?
